airbyte-integrations/connectors/source-chargify/source_chargify/source.py

```diff
@@ -0,0 +1,133 @@
+        if results:
+            url_query = urlparse(response.url).query
+            query_params = parse_qs(url_query)
+
+            new_params = {}
+            for param in query_params:
+                if param == "page":
+                    new_params[param] = int(query_params[param][0]) + 1
+                else:
+                    new_params[param] = query_params[param][0]
+
+            return new_params
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        
+        if next_page_token is None and self.is_first_requests is True:
+            self.is_first_requests == False
+            return {"page": self._page, "per_page": self._per_page}
+
+        return next_page_token
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        TODO: Override this method to define how a response is parsed.
+        :return an iterable containing each record in the response
+        """
+        yield {}
+
+
+class Customers(ChargifyStream):
+
+    # TODO: Fill in the primary key. Required. This is usually a unique field in the stream, like an ID or a timestamp.
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+    
+        return "customers.json"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # Chargify API: https://developers.chargify.com/docs/api-docs/b3A6MTQxMDgyNzY-list-or-find-customers
+        # it returns an Array of Customers objects.
+        customers = response.json()
+        for customer in customers:
+            yield customer["customer"]
+
+
+class Subscriptions(ChargifyStream):
+
+    # TODO: Fill in the primary key. Required. This is usually a unique field in the stream, like an ID or a timestamp.
+    primary_key = "id"
+
+    def path(self, **kwargs) -> str:
+    
+        return "subscriptions.json"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # Chargify API: https://developers.chargify.com/docs/api-docs/b3A6MTQxMDgyNzY-list-or-find-customers
+        # it returns an Array of Customers objects.
+        subscriptions = response.json()
+        for subscription in subscriptions:
+            yield subscription["subscription"]
+
+
+# Source
+class SourceChargify(AbstractSource):
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        try:
+            auth = HTTPBasicAuth(config["api_key"], "X")
+            converted_args = self.convert_config2stream_args(config)
+            customers_gen = Customers(authenticator=auth, **converted_args).read_records(sync_mode=SyncMode.full_refresh)
```

Comment:
```suggestion
            customers_gen = Customers(authenticator=auth, subdomain=config["subdomain"]).read_records(sync_mode=SyncMode.full_refresh)
```
I don't think `convert_config2stream_args`. is useful here.

Reply:
Could you also try to read records from both streams in the check connection? To make sure we can read all the streams defined in the connector.